brave/src/main/java/brave/internal/recorder/MutableEndpoint.java

```diff
@@ -0,0 +1,176 @@
+package brave.internal.recorder;
+
+import brave.internal.Nullable;
+
+/** The network context of a node in the service graph. */
+public final class MutableEndpoint {
```

Comment:
this type might go away, or become package private

Reply:
I could not find a unit test for this class methods, I believe this one specially needs one due to the many ifs. What do you think?

Reply:
yep indeed this needs unit tests.. there area a couple other new methods that do as well. good point